From 47b49169a9d982c6f16a2d7922b3ac6ce045120f Mon Sep 17 00:00:00 2001
From: Richard Zhu <hongxing.zhu@nxp.com>
Date: Mon, 7 May 2018 12:54:40 +0800
Subject: [PATCH 3742/5242] MLK-18210 ARM64: dts: imx8qm: correct the pds of
 pcie

commit  107bd0f46251aa14a3e6a42e14a188e1cf6f65a3 from
https://source.codeaurora.org/external/imx/linux-imx.git

HSIO MSIC/GPIO are powered by the pd_hsio_gpio domain.
Use the pd_hsio_gpio as the parent pd of the imx8 hsio to
make sure that the pd_hsio_gpio domain would be tuend on
when enable HSIO module.
BTW, PHY calibration of the PHYX2_1/PHYX1 is relied on the
results of the PHYX2_0.
So, all the HSIO PDs should be turned on when use PCIe
or SATA.

Signed-off-by: Richard Zhu <hongxing.zhu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi      |   54 ++++++++++----------
 1 file changed, 28 insertions(+), 26 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index 22b49fb..baf9f0b 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -428,49 +428,51 @@
 			#address-cells = <1>;
 			#size-cells = <0>;
 
-			pd_serdes0: PD_HSIO_SERDES_0 {
-				reg = <SC_R_SERDES_0>;
+			pd_hsio_gpio: PD_HSIO_GPIO {
+				reg = <SC_R_HSIO_GPIO>;
 				#power-domain-cells = <0>;
 				power-domains =<&pd_hsio>;
 				#address-cells = <1>;
 				#size-cells = <0>;
 
-				pd_pcie0: PD_HSIO_PCIE_A {
-					reg = <SC_R_PCIE_A>;
+				pd_serdes0: PD_HSIO_SERDES_0 {
+					reg = <SC_R_SERDES_0>;
 					#power-domain-cells = <0>;
-					power-domains =<&pd_serdes0>;
+					power-domains =<&pd_hsio_gpio>;
 					#address-cells = <1>;
 					#size-cells = <0>;
 
-					pd_pcie1: PD_HSIO_PCIE_B {
-						reg = <SC_R_PCIE_B>;
-						#power-domain-cells = <0>;
-						power-domains =<&pd_pcie0>;
-						#address-cells = <1>;
-						#size-cells = <0>;
-
-					pd_serdes1: PD_HSIO_SERDES_1 {
-						reg = <SC_R_SERDES_1>;
+					pd_pcie0: PD_HSIO_PCIE_A {
+						reg = <SC_R_PCIE_A>;
 						#power-domain-cells = <0>;
-						power-domains =<&pd_pcie1>;
+						power-domains =<&pd_serdes0>;
 						#address-cells = <1>;
 						#size-cells = <0>;
 
-						pd_sata0: PD_HSIO_SATA_0 {
-							reg = <SC_R_SATA_0>;
+						pd_pcie1: PD_HSIO_PCIE_B {
+							reg = <SC_R_PCIE_B>;
 							#power-domain-cells = <0>;
-							power-domains =<&pd_serdes1>;
+							power-domains =<&pd_pcie0>;
+							#address-cells = <1>;
+							#size-cells = <0>;
+
+							pd_serdes1: PD_HSIO_SERDES_1 {
+								reg = <SC_R_SERDES_1>;
+								#power-domain-cells = <0>;
+								power-domains =<&pd_pcie1>;
+								#address-cells = <1>;
+								#size-cells = <0>;
+
+								pd_sata0: PD_HSIO_SATA_0 {
+									reg = <SC_R_SATA_0>;
+									#power-domain-cells = <0>;
+									power-domains =<&pd_serdes1>;
+								};
+							};
 						};
 					};
-					};
 				};
 			};
-
-			pd_gpio: PD_HSIO_GPIO {
-				reg = <SC_R_HSIO_GPIO>;
-				#power-domain-cells = <0>;
-				power-domains =<&pd_hsio>;
-			};
 		};
 
 		pd_audio: PD_AUDIO {
@@ -3472,7 +3474,7 @@
 				 <0 0 0 2 &gic 0 74 4>,
 				 <0 0 0 3 &gic 0 75 4>,
 				 <0 0 0 4 &gic 0 76 4>;
-		power-domains = <&pd_pcie0>;
+		power-domains = <&pd_pcie1>;
 		fsl,max-link-speed = <3>;
 		hsio-cfg = <PCIEAX1PCIEBX1SATA>;
 		hsio = <&hsio>;
-- 
1.7.9.5

